packages/patternlab/styleguide/source/assets/js/modules/responsiveTable.js

```diff
@@ -87,106 +72,104 @@ export default (function (window, document, $) {
+    if (isHighZoom()) {
+      if (rt.$stickyHeader) rt.$stickyHeader.hide();
+    } else {
+      setWidths(rt);
+    }
+
     if (reset) {
       responsiveTables[index] = rt;
     } else {
       responsiveTables.push(rt);
       rt.$root[0].addEventListener("scroll", handleTableScroll, true);
     }
-    // Decide what should be showing or stuck.
+
     checkVisibility(rt);
   }
 
-  // Certain other components that stick to the top of the page need to be accounted for.
-  // This calculates the additional offset that a table sticky header should drop down.
   function getAdditionalOffset() {
     let additionalOffset = 0;
-    if (
-      $(".js-scroll-anchors")[0] &&
-      document.documentElement.clientWidth <= 765
-    ) {
+    if ($(".js-scroll-anchors")[0] && document.documentElement.clientWidth <= 765) {
       additionalOffset += $(".js-scroll-anchors").height();
     }
     if (document.documentElement.classList.contains("stickyTOC")) {
-      if (document.documentElement.clientWidth <= 841) {
-        additionalOffset += 75;
-      } else {
-        additionalOffset += 70;
-      }
+      additionalOffset += document.documentElement.clientWidth <= 841 ? 75 : 70;
     }
     return additionalOffset > 0 ? additionalOffset : 0;
   }
 
-  // Based on the scroll position, decide whether or not to show or hide or scroll
-  // or stick the header and scrollbar.
   function checkVisibility(rt) {
     const { index, $table, $stickyHeader, headerStuck } = rt;
 
-      // Handle header visibility.
     if ($stickyHeader && $table.find("thead").length > 0) {
       const stuckTop = $stickyHeader.offset().top;
       const stuckBottom = stuckTop + $stickyHeader.height();
-
-      const $tableHeader = $table.find('thead');
+      const $tableHeader = $table.find("thead");
       const elementTop = $tableHeader.offset().top;
       const tableBottom = elementTop + $table.height();
 
-      if (
-        !headerStuck &&
-        elementTop < stuckTop &&
-        tableBottom > stuckBottom
-      ) {
+      let activeHeader = null;
+      $table.find("tbody tr").each(function () {
+        let rowTop = $(this).offset().top;
+        if (rowTop <= window.scrollY + stuckBottom) {
+          activeHeader = $(this).find("th").text().trim();
+        }
+      });
+
+      if (activeHeader) {
+        $stickyHeader.find("th").text(activeHeader);
+      }
+
+      if (!headerStuck && elementTop < stuckTop && tableBottom > stuckBottom) {
         responsiveTables[index].headerStuck = true;
-        $stickyHeader.css("opacity", 1);
-        $stickyHeader.css("-webkit-box-shadow", "");
-        $stickyHeader.css("box-shadow", "");
-        $stickyHeader.css("pointer-events", "all");
-      } else if (
-        headerStuck &&
-        (elementTop > stuckTop || tableBottom < stuckBottom)
-      ) {
+        $stickyHeader.css({
+          opacity: 1,
+          "pointer-events": "all",
+          "box-shadow": "0px 2px 10px rgba(0, 0, 0, 0.1)",
```

Comment:
If we are adding a box shadow can we keep it consistent with what's in Mayflower, see var `$box-shadow: 0 0.25rem 0.5rem $c-box-shadow-black !default;`
before:
Also there's a border that's missing https://mayflower.digital.mass.gov/patternlab/b/develop/patterns/05-pages-information-details-responsive-table/05-pages-information-details-responsive-table.html
after: 
https://mayflower.digital.mass.gov/patternlab/b/feature/DP-36119_change_sticky_header/patterns/05-pages-information-details-responsive-table/05-pages-information-details-responsive-table.html

Reply:
<img width="1023" alt="Screenshot 2025-03-04 at 12 22 25" src="https://github.com/user-attachments/assets/c978dde7-179f-4a8d-bbfd-6cf9da8eca94" />
Not sure if its cached, but my local has no visual regressions after changes. see screenshot